python/nano/src/bigdl/nano/pytorch/onnx/onnxrt_inference.py

```diff
@@ -0,0 +1,199 @@
+               If backend is NOT set to "onnx", a torch tensor is needed and the pytorch
+               forwarding method will be called.
+        :param batch_size: int, inferencing batch_size. This value should not affect the
+               final inferencing result but will affect resources cost(e.g. memory and time).
+               Default to None, which takes all input_data in one batch.
+        :param file_path: The path to save onnx model file.
+        :param sess_options: ortsess options in ort.SessionOptions type.
+        :param backend: str, to set the backend library. "onnx" for onnxruntime, which
+               provides lower latency and any other value will make `inference` call
+               the pytorch forwarding method.
+        :param **kwargs: any other keywords that will be passed to onnx session's building.
+        '''
+
+        if backend == "onnx":
+            if not self._ortsess_up_to_date:
+                warnings.warn("Onnxruntime session will be built implicitly,"
+                              " this may harm your inference latency.")
+                input_sample = torch.Tensor(input_data)
+                self._build_ortsess(input_sample=input_sample,
+                                    file_path=file_path,
+                                    sess_options=sess_options,
+                                    **kwargs)
+            input_name = self._ortsess.get_inputs()[0].name
+            if batch_size is None:
+                # this branch is only to speed up the inferencing when batch_size is set to None.
+                ort_inputs = {input_name: input_data}
+                ort_outs = self._ortsess.run(None, ort_inputs)
+                return ort_outs[0]
+            else:
+                yhat_list = []
+                sample_num = input_data.shape[0]  # the first dim should be sample_num
+                batch_num = math.ceil(sample_num / batch_size)
+                for batch_id in range(batch_num):
+                    ort_inputs = {input_name: input_data[batch_id * batch_size:
+                                                         (batch_id + 1) * batch_size]}
+                    ort_outs = self._ortsess.run(None, ort_inputs)
+                    yhat_list.append(ort_outs[0])
+                # this operation may cause performance degradation
+                yhat = np.concatenate(yhat_list, axis=0)
+                return yhat
+        else:
+            # inference w/o onnxruntime (fallback to pytorch native forward)
+            self.eval()
+            with torch.no_grad():
+                yhat_list = []
+                sample_num = input_data.shape[0]  # the first dim should be sample_num
+                batch_size = batch_size if batch_size else sample_num
+                batch_num = math.ceil(sample_num / batch_size)
+                for batch_id in range(batch_num):
+                    yhat_list.append(self(input_data[batch_id * batch_size:
```

Comment:
shall we call forward or predict_step here?

Reply:
I won't use `predict_step` here because `predict_step`  is not always defined in users code. So I use `self()` to call `forward` here.